aten/src/ATen/native/cuda/TensorTopK.cu

```diff
@@ -162,7 +172,349 @@ __global__ void gatherTopK(at::cuda::detail::TensorInfo<T, IndexType> input,
+  T* kthValues = reinterpret_cast<T*>(kthValues_buffer.get());
+  AT_CUDA_CHECK(cudaMemsetAsync(kthValues, 0, numInputSlices * sizeof(T), c10::cuda::getCurrentCUDAStream()));
+
+  auto semaphores_buffer = allocator.allocate(numInputSlices * sizeof(int));
+  int* semaphores = reinterpret_cast<int*>(semaphores_buffer.get());
+  AT_CUDA_CHECK(cudaMemsetAsync(semaphores, 0, numInputSlices * sizeof(int), c10::cuda::getCurrentCUDAStream()));
+
+  auto ksToFind_buffer = allocator.allocate(numInputSlices * sizeof(IndexType));
+  IndexType* ksToFind = reinterpret_cast<IndexType*>(ksToFind_buffer.get());
+  fill<IndexType><<<std::min((numInputSlices + 511) / 512, (IndexType)65535), 512, 0, c10::cuda::getCurrentCUDAStream()>>>(
+          ksToFind, outputSliceSize, numInputSlices);
+  C10_CUDA_KERNEL_LAUNCH_CHECK();
+
+  auto desired_buffer = allocator.allocate(numInputSlices * sizeof(Bitwise));
+  Bitwise* desired = reinterpret_cast<Bitwise*>(desired_buffer.get());
+
+  auto counts_buffer = allocator.allocate(num_blocks * RADIX_DIGITS * sizeof(IndexType));
+  IndexType* counts = reinterpret_cast<IndexType*>(counts_buffer.get());
+
+  Bitwise desiredMask = 0;
+  dim3 grid;
+  TORCH_INTERNAL_ASSERT(getGridFromTiles(num_blocks, grid), "Too many slices to sort");
+  dim3 block(BLOCK_THREADS);
+
+#define RUN_K(BIT)                                             \
+  radixFindKthValues<T, IndexType, Bitwise, Dim, Order, BIT>   \
+      <<<grid, block, 0, c10::cuda::getCurrentCUDAStream()>>>( \
+          input,                                               \
+          inputSliceSize,                                      \
+          ksToFind,                                            \
+          numInputSlices,                                      \
+          inputWithinSliceStride,                              \
+          current_bit,                                         \
+          blocks_per_slice,                                    \
+          desiredMask,                                         \
+          semaphores,                                          \
+          desired,                                             \
+          counts,                                              \
+          kthValues);                                          \
+  C10_CUDA_KERNEL_LAUNCH_CHECK();
+
+#define RUN_BIT()                                              \
+  if (radix_bits == 6) {                                       \
+    RUN_K(6);                                                  \
+  } else if (radix_bits == 4) {                                \
+    RUN_K(4);                                                  \
+  } else if (radix_bits == 2) {                                \
+    RUN_K(2);                                                  \
+  } else {                                                     \
+    TORCH_INTERNAL_ASSERT(                                     \
+        false, "RADIX_BIT ", radix_bits, " is not supported"); \
+  }
```

Comment:
Is it possible to make `radix_bits` not a template argument? Hopefully, this will help save some compilation time and binary size.